app/models/application_form.rb

```diff
@@ -708,7 +708,17 @@ def can_submit?
 
   def meets_conditions_for_adviser_interruption?
     eligible_to_sign_up_for_a_teaching_training_adviser? &&
-      adviser_interruption_response != false
+      adviser_interruption_response != false && FeatureFlag.active?(:adviser_sign_up)
+  end
+
+  def prefill_preferred_teaching_subject_id
```

Comment:
I think these three new methods belong on the interruption form where they are used. It feels like a very specific use case.